src/stan/io/reader.hpp

```diff
@@ -1227,116 +1317,351 @@ class reader {
       v(i) = scalar_ub_constrain(ub);
+    }
     return v;
   }
 
   template <typename TU>
-  inline vector_t vector_ub_constrain(const TU ub, size_t m, T &lp) {
+  inline vector_t vector_ub_constrain(const TU ub, size_t m, T& lp) {
     vector_t v(m);
-    for (size_t i = 0; i < m; ++i)
+    for (size_t i = 0; i < m; ++i) {
       v(i) = scalar_ub_constrain(ub, lp);
+    }
     return v;
   }
 
   template <typename TU>
   inline row_vector_t row_vector_ub(const TU ub, size_t m) {
     row_vector_t v(m);
-    for (size_t i = 0; i < m; ++i)
+    for (size_t i = 0; i < m; ++i) {
       v(i) = scalar_ub(ub);
+    }
     return v;
   }
 
   template <typename TU>
   inline row_vector_t row_vector_ub_constrain(const TU ub, size_t m) {
     row_vector_t v(m);
-    for (size_t i = 0; i < m; ++i)
+    for (size_t i = 0; i < m; ++i) {
       v(i) = scalar_ub_constrain(ub);
+    }
     return v;
   }
 
   template <typename TU>
-  inline row_vector_t row_vector_ub_constrain(const TU ub, size_t m, T &lp) {
+  inline row_vector_t row_vector_ub_constrain(const TU ub, size_t m, T& lp) {
     row_vector_t v(m);
-    for (size_t i = 0; i < m; ++i)
+    for (size_t i = 0; i < m; ++i) {
       v(i) = scalar_ub_constrain(ub, lp);
+    }
     return v;
   }
 
   template <typename TU>
-  inline matrix_t matrix_ub(const TU ub, size_t m, size_t n) {
-    matrix_t v(m, n);
-    for (size_t j = 0; j < n; ++j)
-      for (size_t i = 0; i < m; ++i)
+  inline matrix_t matrix_ub(const TU ub, size_t n, size_t m) {
+    matrix_t v(n, m);
+    for (size_t j = 0; j < m; ++j) {
+      for (size_t i = 0; i < n; ++i) {
         v(i, j) = scalar_ub(ub);
+      }
+    }
     return v;
   }
 
   template <typename TU>
-  inline matrix_t matrix_ub_constrain(const TU ub, const size_t m, size_t n) {
-    matrix_t v(m, n);
-    for (size_t j = 0; j < n; ++j)
-      for (size_t i = 0; i < m; ++i)
+  inline matrix_t matrix_ub_constrain(const TU ub, const size_t n, size_t m) {
+    matrix_t v(n, m);
+    for (size_t j = 0; j < m; ++j) {
+      for (size_t i = 0; i < n; ++i) {
         v(i, j) = scalar_ub_constrain(ub);
+      }
+    }
     return v;
   }
 
   template <typename TU>
-  inline matrix_t matrix_ub_constrain(const TU ub, const size_t m, size_t n,
-                                      T &lp) {
-    matrix_t v(m, n);
-    for (size_t j = 0; j < n; ++j)
-      for (size_t i = 0; i < m; ++i)
+  inline matrix_t matrix_ub_constrain(const TU ub, const size_t n, size_t m,
+                                      T& lp) {
+    matrix_t v(n, m);
+    for (size_t j = 0; j < m; ++j) {
+      for (size_t i = 0; i < n; ++i) {
         v(i, j) = scalar_ub_constrain(ub, lp);
+      }
+    }
     return v;
   }
 
+  /**
+   * Return a sparse matrix of the specified dimensionality and number of
+   * nonzero elements made up of the next scalars arranged in
+   * column-major order. Throws if any scalar value is less than ub.
+   *
+   * @tparam TL the type of the upper bound.
```

Comment:
I strongly prefer `T`, but if there's going to be more informative letters attached, they need to match.  So presumably this should be `TU` for upper bound?